src/lib/project-saver-hoc.jsx

```diff
@@ -118,25 +119,36 @@ const ProjectSaverHOC = function (WrappedComponent) {
          * @param {?object} requestParams - object of params to add to request body
          */
         storeProject (projectId, requestParams) {
-            return this.props.vm.saveProjectSb3()
-                .then(content => {
-                    const assetType = storage.AssetType.Project;
-                    const dataFormat = storage.DataFormat.SB3;
-                    const body = new FormData();
-                    body.append('sb3_file', content, 'sb3_file');
-                    if (requestParams) {
-                        for (const key in requestParams) {
-                            body.append(key, requestParams[key]);
-                        }
-                    }
-                    // when id is undefined or null, storage.store as we have
-                    // configured it will create a new project with id
-                    return storage.store(
-                        assetType,
-                        dataFormat,
-                        body,
-                        projectId
-                    );
+            requestParams = requestParams || {};
+            return Promise.all(this.props.vm.assets
```

Comment:
Does this assets saving make sense to do in the case where we don't own the current project loaded in the vm? Seems like we're putting the cart before the horse here. I suspect we only want to do this after we've successfully saved the project file, or if we know we own the project.

Reply:
Imagine you save the JSON, and then all the assets fail to save. Now your JSON is pointing to a bunch of MD5s that don't exist, so you've effectively corrupted your project. To avoid that, we want to make sure all the assets are stored safely before attempting to save the project JSON. Any asset save failure should cause the whole project save process to fail. And, we don't really care who owns the project as far as the assets go. You just have to be authenticated.

As far as who owns the project, I would hope that's already been determined once we get here? All I'm doing is changing the mechanics of how the project data is saved, not anything to do with ownership. All of that should remain as it was.

Reply:
I think I'm understanding better.

Let me clarify. The case where you don't own the project is basically when you're creating a project, from one already loaded in the vm. In that case, we'll get here and start saving assets even though there is not yet any project that you own that's associated with them.

But that's OK! Right? Before, I was assuming that assets are stored with some kind of reference to the project that owns them... but that's not true, right? (The project references the assets, but not the other way around.) If I'm understanding this right, then I totally agree with you that we should save assets, and only afterwards save the project JSON.

Reply:
Yes exactly.